app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -132,34 +132,43 @@ class ConversationViewModel @Inject constructor(
     }
 
     private fun listenConversationDetails() = viewModelScope.launch {
-        observeConversationDetails(conversationId)
-            .filterIsInstance<ObserveConversationDetailsUseCase.Result.Success>() // TODO handle StorageFailure
-            .map { it.conversationDetails }
-            .collect { conversationDetails ->
-                val conversationName = when (conversationDetails) {
-                    is ConversationDetails.OneOne -> conversationDetails.otherUser.name.orEmpty()
-                    else -> conversationDetails.conversation.name.orEmpty()
-                }
-                val conversationAvatar = when (conversationDetails) {
-                    is ConversationDetails.OneOne ->
-                        ConversationAvatar.OneOne(conversationDetails.otherUser.previewPicture?.let {
-                            UserAvatarAsset(wireSessionImageLoader, it)
-                        })
-                    is ConversationDetails.Group ->
-                        ConversationAvatar.Group(conversationDetails.conversation.id)
-                    else -> ConversationAvatar.None
-                }
-                val conversationDetailsData = when (conversationDetails) {
-                    is ConversationDetails.Group -> ConversationDetailsData.Group(conversationDetails.conversation.id)
-                    is ConversationDetails.OneOne -> ConversationDetailsData.OneOne(conversationDetails.otherUser.id)
-                    else -> ConversationDetailsData.None
+        withContext(dispatchers.io()) {
+            observeConversationDetails(conversationId)
+                .collect { result ->
+                    when (result) {
+                        is Failure -> navigateToHome()
+                        is Success -> emitConversationDetails(result.conversationDetails)
+                    }
                 }
-                conversationViewState = conversationViewState.copy(
-                    conversationName = conversationName,
-                    conversationAvatar = conversationAvatar,
-                    conversationDetailsData = conversationDetailsData
+        }
+    }
+
+    private fun emitConversationDetails(conversationDetails: ConversationDetails) {
```

Comment:
This method doesn't emit anything so I think maybe "handleConversationDetails" would be better